frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/DivergedFromRemoteDialog.kt

```diff
@@ -0,0 +1,104 @@
+  private val remoteBranch: IRemoteTrackingBranchReference,
+  private val branch: IManagedBranchSnapshot,
+  private val relationToRemote: SyncToRemoteStatus
+) : DialogWrapper(project, /* canBeParent */ true) {
+
+  private var divergenceResolutionOption = DivergenceResolutionOption.DO_NOT_SYNC
+
+  init {
+    title =
+      getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.title")
+    setOKButtonMnemonic('O'.code)
+    setCancelButtonText(getString("action.GitMachete.BaseTraverseAction.dialog.cancel-traverse"))
+    super.init()
+  }
+
+  fun showAndGetThePreferredAction() =
+    if (showAndGet()) {
+      divergenceResolutionOption
+    } else {
+      null
+    }
+
+  override fun createCenterPanel() = panel {
+    rowCompat {
+      label(
+        fmt(
+          getString(
+            "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.text.HTML"
+          ),
+          branch.name,
+          remoteBranch.name,
+          getRelationToRemoteDescription()
+        )
+      )
+    }
+    buttonsGroupCompat {
+      rowCompat {
+        radioButton(
+          getString(
+            "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.do-not-resolve-option"
+          ),
+          DivergenceResolutionOption.DO_NOT_SYNC
+        ).commentCompat(getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.do-not-resolve-option.comment"))
+      }
+      when (relationToRemote) {
+        SyncToRemoteStatus.DivergedFromAndOlderThanRemote -> rowCompat {
+          radioButton(
+            getString(
+              "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.reset-option"
+            ),
+            DivergenceResolutionOption.RESET_TO_REMOTE
+          ).commentCompat(getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.reset-on-remote.comment"))
+        }
+        SyncToRemoteStatus.DivergedFromAndNewerThanRemote ->
+          rowCompat {
+            radioButton(
+              getString(
+                "action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.force-push-option"
+              ),
+              DivergenceResolutionOption.FORCE_PUSH
+            ).commentCompat(getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.force-push.comment"))
+          }
+
+        else -> {}
+      }
+    }
+      .bind(
+        MutableProperty(::divergenceResolutionOption) { divergenceResolutionOption = it },
+        DivergenceResolutionOption::class.java
+      )
+  }
+
+  private fun getRelationToRemoteDescription(): String =
+    when (relationToRemote) {
+      SyncToRemoteStatus.DivergedFromAndNewerThanRemote ->
+        getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.older-than-remote")
+      SyncToRemoteStatus.DivergedFromAndOlderThanRemote ->
+        getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-remote.newer-than-remote")
```

Comment:
Whoops the messages are swapped 😅

Reply:
actually, the message keys are swapped but the underlying values are correct for each case XD